.travis.yml

```diff
@@ -0,0 +1,45 @@
+language: go
+
+go:
+  - "1.9.7"
+  - "1.10.x"
+  - "1.11.x"
+  - "1.12.x"
+  - "1.x"
+  - master
+  
+os:
+  - linux
+  - osx
+
+cache:
+  directories:
+    - $HOME/.cache/go-build
+    - $HOME/gopath/pkg/mod
+    
+# Skip the installation step. Don't "go get" dependencies.
+install: true
+
+env:
+  global:
+    - GO111MODULE=on
+
+# Only clone the most recent commit.
+git:
+  depth: 1
+
+before_script:
+  # needed until we stop supporting 1.10.x
+  GO111MODULE=off;
+  go get -t -d ./...;   
+
+script:
+  - ./.travis.gofmt.sh
+  - go generate -x ./... && git diff --exit-code; code=$?; git checkout -- .; (exit $code) # Check that go generate ./... produces a zero diff; clean up any changes afterwards.
+  - go vet ./...
+  - go test -v -race ./...
+  - go test -v ./...
```

Comment:
Do we need to run tests both with and without `-race`? Can we run only `go test -v -race ./...`?

Reply:
Looks like the second call is redundant, will remove